.github/workflows/makefile.yml

```diff
@@ -0,0 +1,280 @@
+name: Makefile CI
+
+on:
+  push:
+  pull_request:
+env:
+  # BASIC_PACKAGES is used fo the basic tools (make)
+  BASIC_PACKAGES: git build-essential ccache libgc-dev graphviz libunwind-dev libreadline-dev pkg-config
+  # MORE_PACKAGES is used for more tools and libs (make more)
+  MORE_PACKAGES: libgmp-dev libcurl4-openssl-dev libevent-dev openjdk-17-jdk-headless libgles-dev libegl-dev libsdl1.2-dev libsdl2-dev libsdl2-mixer-dev libsdl2-gfx-dev libsdl2-image-dev libsdl2-net-dev libsdl2-ttf-dev libgtk-3-dev inkscape postgresql libxdg-basedir-dev gettext libsqlite3-dev libpq-dev libmongoc-dev openmpi-bin libopenmpi-dev
+  JAVA_HOME: /usr/lib/jvm/java-17-openjdk-amd64/
+  JNI_LIB_PATH: /usr/lib/jvm/java-17-openjdk-amd64/lib/server/
+  LD_LIBRARY_PATH: /usr/lib/jvm/java-17-openjdk-amd64/lib/server/
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: sudo apt-get update && sudo apt-get install -y $BASIC_PACKAGES
+    - run: make
+    - uses: actions/upload-artifact@v4
+      with:
+        path: bin
+        name: bin
+
+  sanity-check:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - run: misc/jenkins/ensuregit.sh
+    - run: misc/jenkins/checkwhitespaces.sh
+      if: success() || failure()
```

Comment:
The docs suggest to use `!cancelled()` to run a job or step regardless of its success or failure

https://docs.github.com/en/actions/learn-github-actions/expressions#always